antarest/study/storage/variantstudy/variant_study_service.py

```diff
@@ -1050,3 +1054,90 @@ def initialize_additional_data(self, variant_study: VariantStudy) -> bool:
                 exc_info=e,
             )
             return False
+
+    def clear_all_snapshots(self, limit: int, params: t.Optional[RequestParameters]) -> str:
+        """
+        Clear all variant snapshots older than `limit` (in hours).
+        Only available for admin users.
+        `limit` must be a positive integer.
+
+        Args:
+            limit (integer): number of hours
+            params: request parameters used to identify the user status
+        Returns: None
+
+        Raises:
+            UserHasNotPermissionError
+            VariantAgeMustBePositive
+        """
+        if params is None:
```

Comment:
we should also check that `user.is_site_admin()`, to allow only admins to use this feature